fvm/meter/weighted/meter.go

```diff
@@ -75,6 +197,14 @@ func WithComputationWeights(weights ExecutionEffortWeights) WeightedMeterOptions
 
 // WithMemoryWeights sets the weights for the memory intensities
 func WithMemoryWeights(weights ExecutionMemoryWeights) WeightedMeterOptions {
+	if int(common.MemoryKindLast)-len(weights) != 1 {
+		return func(m *Meter) {
+			for kind, weight := range weights {
+				m.memoryWeights[kind] = weight
```

Comment:
I think this will overwrite `DefaultMemoryWeights`, making this unexpectedly destructive. I don't think that will ever matter in production but might mess up tests.